extensions/amp-story/0.1/media-pool.js

```diff
@@ -42,6 +44,67 @@ export const MediaType = {
 export let ElementDistanceFnDef;
 
 
+/**
+ * Represents a task to be executed on a media element.
+ * @typedef {function(!HTMLMediaElement): !Promise}
+ */
+let ElementTaskDef;
+
+
+/**
+ * "Blesses" the specified media element for future playback without a user
+ * gesture.  In order for this to bless the media element, this function must
+ * be invoked in response to a user gesture.
+ * @param {!HTMLMediaElement} mediaEl The media element to bless.
+ * @return {!Promise} A promise that is resolved when blessing the media
+ *     element is complete.
+ */
+function blessMediaElement(mediaEl) {
+  const isPaused = mediaEl.paused;
+  const isMuted = mediaEl.muted;
+  const currentTime = mediaEl.currentTime;
+
+  /**
+   * @return {!Promise} A promise that is resolved when playback has been
+   *    initiated or rejected if playback fails to initiate.  If the media
+   *    element is already playing, the promise is immediately resolved
+   *    without playing the media element again, to avoid interrupting
+   *    playback.
+   */
+  const playFn = () => {
+    if (isPaused) {
+      // The playFn() invocation is wrapped in a Promise.resolve(...) due to
+      // the fact that some browsers return a promise from media elements'
+      // play() function, while others return a boolean.
+      return Promise.resolve(mediaEl.play());
+    }
+
+    // This media element was already playing.
+    return Promise.resolve();
+  };
+
+  const blessPromise = playFn().then(() => {
+    mediaEl.muted = false;
+
+    if (isPaused) {
+      mediaEl.pause();
+      mediaEl.currentTime = currentTime;
+    }
+
+    if (isMuted) {
+      mediaEl.muted = true;
+    }
+  });
+
+  blessPromise.catch(reason => {
+    dev().expectedError('AMP-STORY', 'Blessing media element failed:',
+        reason, mediaEl);
+  }).then(() => dispatch(mediaEl, 'bless', false));
```

Comment:
Might be worth adding a comment to mention this line works as a `finally` and is triggered on both success or failure of the promise

Reply:
refactored this